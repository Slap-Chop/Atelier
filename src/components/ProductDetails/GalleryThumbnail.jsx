import React from 'react';
import noImage from './Images/noImage.png';

const GalleryThumbnail = (props) => {
  return (
    <div
    className='galleryThumbnailContainer'
      >
      <img
        className='galleryThumbnail'
        onClick={() => {
          props.click(props.index)
        }}
        style={{opacity: props.selected}}
      src={`${props.photo.thumbnail_url}` || noImage}
      alt='Image N/A'/>

    </div>
  )
}

export default GalleryThumbnail;