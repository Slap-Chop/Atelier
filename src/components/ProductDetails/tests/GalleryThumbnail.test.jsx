import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import GalleryThumbnail from '../GalleryThumbnail.jsx';

const mockFunction = jest.fn();
const testThumbnail = shallow(<GalleryThumbnail click={mockFunction}
photo={{thumbnail_url: 'https://images.unsplash.com/photo-1534011546717-40…hcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80', url: 'https://images.unsplash.com/photo-1534011546717-40…cHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2734&q=80'}}/>)

test('Icons should render properly', () => {
  expect(toJson(testThumbnail)).toMatchSnapshot();
})

test('Clicking the icon image should call the function passed from props', () => {
  testThumbnail.find('img.galleryThumbnail').simulate('click');
  expect(mockFunction).toHaveBeenCalled();
})