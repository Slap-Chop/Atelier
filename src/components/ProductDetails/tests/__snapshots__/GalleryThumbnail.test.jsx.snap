// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons should render properly 1`] = `
<div
  className="galleryThumbnailContainer"
>
  <img
    className="galleryThumbnail"
    onClick={[Function]}
    src="https://images.unsplash.com/photo-1534011546717-40â€¦hcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
    style={
      Object {
        "opacity": undefined,
      }
    }
  />
</div>
`;
